staging/src/k8s.io/cloud-provider/controllers/service/controller.go

```diff
@@ -743,7 +743,7 @@ func (s *Controller) lockedUpdateLoadBalancerHosts(service *v1.Service, hosts []
 }
 
 func wantsLoadBalancer(service *v1.Service) bool {
-	return service.Spec.Type == v1.ServiceTypeLoadBalancer
+	return service.Spec.Type == v1.ServiceTypeLoadBalancer && service.Spec.LoadBalancerClassName == ""
```

Comment:
service.Spec.LoadBalancerClassName != nil && service.Spec.LoadBalancerClassName != ""

Reply:
I think there is no need to check the empty string since it can't pass `DNS1123Label` validation?